homeassistant/components/modbus/validators.py

```diff
@@ -68,6 +70,17 @@ def sensor_schema_validator(config):
             f"but {config[CONF_COUNT]} registers have a size of {bytecount} bytes"
         )
 
+    if config.get(CONF_REGISTER_SIZE, 2) > 2:
+        if (
+            config[CONF_DATA_TYPE] == DATA_TYPE_INT
```

Comment:
What about float ? What about  a struct like e.g. “>4i”, those are affected by the register_size change as well. 

Eliminating swap is a bad idea, because that means your solution will not work on little indian processors (network format is big indian).

If you want to add a different register size you should support all standard configurations.